src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -80,9 +82,11 @@ private void registerConnectionsGauge() {
     }
   }
 
-  protected JmxProxy connectImpl(Node node, int connectionTimeout) throws ReaperException, InterruptedException {
+  protected JmxProxy connectImpl(Node node, AppContext context) throws ReaperException, InterruptedException {
     // use configured jmx port for host if provided
-    String host = node.getHostname();
+    String host = context.config.isInSidecarMode()
+        ? LOCALHOST
+        : node.getHostname();
```

Comment:
change this to `Preconditions.checkArgument(!context.config.isInSidecarMode() || LOCALHOST.equals(node.getHostname()));`
 so we smoke out all the places that are trying to make jmx connections to other nodes (there's a few places in the code!)